handlers/resources_test.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Alex Ellis 2017. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package handlers
+
+import (
+	"testing"
+
+	"github.com/openfaas/faas/gateway/requests"
+)
+
+// Test_ParseMemory exploratory testing to document how to convert
+// from Docker limits notation to bytes value.
+func Test_ParseMemoryInMegabytes(t *testing.T) {
+	value := "512 m"
+
+	val, err := parseMemory(value)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if val != megaBytes(512) {
+		t.Errorf("want: %d got: %d", 1024, val)
+	}
+}
+
+func megaBytes(mbs int64) int64 {
+	return 1024 * 1024 * mbs
+}
+
+func Test_ParseInvalidMemoryInMegabytes(t *testing.T) {
+	req := requests.CreateFunctionRequest{
+		Requests: &requests.FunctionResources{
+			Memory: "wrong",
+		},
+		Limits: &requests.FunctionResources{},
+	}
+
+	res := buildResources(&req)
+
+	if res.Reservations != nil {
+		t.Fatalf("Expected nil reservation due to incorrect value provided")
+	}
+}
+
```

Comment:
Could add the following tests for increased coverage:

```

func TestBuildSwarmResourcesAddsMemoryLimits(t *testing.T) {
	req := requests.CreateFunctionRequest{
		Requests: &requests.FunctionResources{},
		Limits: &requests.FunctionResources{
			Memory: "512m",
		},
	}

	res := buildResources(&req)

	if res.Limits.MemoryBytes != megaBytes(512) {
		t.Fatalf("Expected Memory limit of 512m, got %d", res.Limits.MemoryBytes)
	}
}

func TestBuildSwarmResourcesAddsMemoryReservations(t *testing.T) {
	req := requests.CreateFunctionRequest{
		Requests: &requests.FunctionResources{
			Memory: "512m",
		},
		Limits: &requests.FunctionResources{},
	}

	res := buildResources(&req)

	if res.Reservations.MemoryBytes != megaBytes(512) {
		t.Fatalf("Expected Memory limit of 512m, got %d", res.Limits.MemoryBytes)
	}
}
```

Reply:
Do they increase the -cover stats?